scipy/sparse/data.py

```diff
@@ -64,8 +64,14 @@ def __itruediv__(self, other):  # self /= other
         else:
             return NotImplemented
 
-    def astype(self, t):
-        return self._with_data(self._deduped_data().astype(t))
+    def astype(self, dtype, casting='unsafe', copy=True):
+        dtype = np.dtype(dtype)
+        if self.dtype != dtype:
+            return self._with_data(self._deduped_data().astype(dtype, casting=casting, copy=copy))
```

Comment:
Done. `copy = False` would be even better here, a copy is already done by `astype`. Maybe I should remove `copy=copy` in the `astype` call for clarification here because a copy is always done here.